src/core/config/srp_client.h

```diff
@@ -225,8 +227,22 @@
  * delay timer.
  *
  */
-#ifndef OPENTHREAD_CONFIG_SRP_CLIENT_UPDATE_TX_DELAY
-#define OPENTHREAD_CONFIG_SRP_CLIENT_UPDATE_TX_DELAY 10
+#ifndef OPENTHREAD_CONFIG_SRP_CLIENT_UPDATE_TX_MIN_DELAY
+#define OPENTHREAD_CONFIG_SRP_CLIENT_UPDATE_TX_MIN_DELAY 10
+#endif
+
+/**
+ * @def OPENTHREAD_CONFIG_SRP_CLIENT_UPDATE_TX_MIN_DELAY
+ *
+ * Specifies the maximum value (in msec) for the short random delay wait time before sending an update message.
+ *
+ * The random delay is chosen uniformly from the min `OPENTHREAD_CONFIG_SRP_CLIENT_UPDATE_TX_MIN_DELAY` up to max value.
+ *
+ * See `OPENTHREAD_CONFIG_SRP_CLIENT_UPDATE_TX_MIN_DELAY` for more details.
+ *
+ */
+#ifndef OPENTHREAD_CONFIG_SRP_CLIENT_UPDATE_TX_MAX_DELAY
+#define OPENTHREAD_CONFIG_SRP_CLIENT_UPDATE_TX_MAX_DELAY 160
```

Comment:
Just curious - how did you pick 160ms? :)

Reply:
I think we might prefer a larger value, e.g. 3000ms. 

The delay is especially meaningful in cases like OMR prefix changing, where all devices need to send srp update at the same time.

Reply:
I went for 150ms randomness interval [10-160]. No particular reason (keeping it short to not add too much delay while allowing for some randomness).

I suggest we keep this relatively short as this will be added every time SRP update is sent. I think there are other mechanisms which inject random timing. For example, network data will need to be disturbed across mesh and not all nodes will get at the same time. Also we can rely on MAC layer CSMA and other delay mechanisms.

Reply:
3 seconds could be too long to be user observable - It's not a good experience that somehow the user can't turn on their light for 3 seconds (or longer).

Reply:
> 3 seconds could be too long to be user observable - It's not a good experience that somehow the user can't turn on their lights for 3 seconds (or longer).

Note that the full IPv6 connectivity (e.g. adding OMR prefix, on-link Prefix)  already needs around 15 seconds to ramp up.

Reply:
Updated the default max delay in the PR to 700 msec. I think that's a good balance (long enough to allow randomness but not too long to add too much delay). As example of other similar constants we use the 750 msec for Parent Response window.

Reply:
@simonlingoogle @wgtdkp any further thoughts on this?

Reply:
700ms SGTM.